src/light_client.rs

```diff
@@ -57,7 +58,13 @@ pub async fn run(
 					}
 					let commitment = header.extrinsics_root.commitment.clone();
 
-					let positions = rpc::generate_random_cells(max_rows, max_cols);
+					let cell_count: u32;
+					if confidence == 100f64 {
+						cell_count = -((1f64 - 99.9f64 / 100f64).log2()).ceil() as u32;
+					} else {
+						cell_count = (-((1f64 - confidence / 100f64).log2())).ceil() as u32;
+					}
```

Comment:
Can we also move code to separate function, e.g. `cell_count_for_confidence` or something better?

Reply:
Actually, already mentioned :D